R/drive_share.R

```diff
@@ -58,31 +65,59 @@ drive_share <- function(file, role = NULL, type = NULL, ..., verbose = TRUE) {
     )
   }
 
+  params <- list(...)
+  params[["role"]] <- role
+  params[["type"]] <- type
+  params[["fields"]] <- "*"
+  permissions_resource <- purrr::map(file$id,
+                                     drive_share_one,
+                                     params = params,
+                                     verbose = verbose)
+  file[["permissions_resource"]] <- NULL
+  file <- tibble::add_column(file, permissions_resource = permissions_resource, .after = 1)
```

Comment:
It is consistent with what happens when we identify files by path. Since we have path (and the user obviously likes to think about them), we include `path` as a column. It sounds sensible to include `permissions_resource` here, since you've got it, and the user is obviously thinking about permissions.